parser/parse_test.go

```diff
@@ -159,8 +161,15 @@ func TestBuilder(t *testing.T) {
 
 				type Enumeration string
 				const (
-					EnumSymbol Enumeration = "enumSymbol"
+					EnumSymbol Enumeration = "enumSymbolValue"
 				)
+
+				type Bytes int
+				const (
+					TenBytes Bytes = 10
+				)
+
```

Comment:
can we test some more complicated cases like `const foo = []byte("foo")` or `const bar = float64(1)`?

Reply:
I'm slightly concerned that this feature will be hard to use when it's a cast, I don't know what these will print.

Also what about:

```
const foo = 1
const bar = foo+1

const (
  x = iota
  y
  z
)
```

Reply:
The `[]byte("foo")` isn't a legal Go constant, so no need to worry about that case. The simple addition and `iota` case work as you probably expect, which is great.

The const [Value](https://golang.org/pkg/go/constant/#Value) for the floating point values surprised me a lot and is different between String and ExactString (see diff below which is from switching to String):

```
    parse_test.go:330: Constant mismatch:   map[string]*types.Type{
          	"ConstExpr": &{
          		... // 1 ignored and 7 identical fields
          		Methods:    nil,
          		Signature:  nil,
        - 		ConstValue: &"(1 + -6.5751i)",
        + 		ConstValue: &"(1 + -65751/10000i)",
          	},
          	"ConstFloat": &{
          		... // 1 ignored and 7 identical fields
          		Methods:    nil,
          		Signature:  nil,
        - 		ConstValue: &"7.8",
        + 		ConstValue: &"8782019273372467/1125899906842624",
          	},
          	"ConstHundred":  &{Name: types.Name{Package: "base/foo/proto", Name: "ConstHundred"}, Kind: "DeclarationOf", ConstValue: &"100"},
          	"ConstNineNine": &{Name: types.Name{Package: "base/foo/proto", Name: "ConstNineNine"}, Kind: "DeclarationOf", ConstValue: &"99"},
          	... // 4 identical entries
          }
```

I'd say the options are
1. Use `ExactString` always
1. Use `ExactString` if the underlying type is a string, and `String` otherwise
1. Use `String` always

(1) is the most accurate but if anyone actually needs the numeric constants it will be confusing. (3) is out because we definitely don't want strings to be possibly truncated. (2) is Ok for documentation purposes, but maybe not for other uses (not sure what they would be though).

Reply:
Let's go with option 2 :/ I knew go represents constants with theoretically infinite precision, but I did not expect it to work like this.

Reply:
OK, I implemented (2) and added some comments. Note that value for string constants is no longer quoted (`ExactString` just calls `strconv.Quote` anyway).